pulpcore/cli/ansible/distribution.py

```diff
@@ -0,0 +1,133 @@
+from typing import Optional, Dict, Any
+
+import click
+
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleDistributionContext,
+    PulpAnsibleRepositoryContext,
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "distribution_type",
+    type=click.Choice(["ansible"], case_sensitive=False),
+    default="ansible",
+)
+@pass_pulp_context
+@click.pass_context
+def distribution(ctx: click.Context, pulp_ctx: PulpContext, distribution_type: str) -> None:
+    if distribution_type == "ansible":
+        ctx.obj = PulpAnsibleDistributionContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+distribution.add_command(list_entities)
+distribution.add_command(show_by_name)
+distribution.add_command(destroy_by_name)
+
+
+@distribution.command()
+@click.option("--name", required=True)
+@click.option("--base-path", required=True)
+@click.option("--repository", help="name of repository")
+@click.option("--version", help="version of repository, leave blank for always latest")
+@click.option("--content-guard")
+@pass_entity_context
+@pass_pulp_context
+def create(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    name: str,
+    base_path: str,
+    repository: Optional[str],
+    version: Optional[str],
+    content_guard: Optional[str],
+) -> None:
+    body: Dict[str, Any] = {"name": name, "base_path": base_path}
+    repo: Dict[str, Any] = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
+    if version:
+        body["repository_version"] = f'{repo["versions_href"]}{version}/'
+    elif repository:
+        body["repository"] = repo["pulp_href"]
+    if content_guard:
+        body["content_guard"] = content_guard
+    result = distribution_ctx.create(body=body)
+    distribution = distribution_ctx.show(result["created_resources"][0])
+    pulp_ctx.output_result(distribution)
+
+
+@distribution.command()
+@click.option("--href", required=True)
```

Comment:
We always try to handle objects by their name, not their href.
Try to focus on the admin, not the api.

Reply:
Hmmm, so for updates should the cli have an option for --new-name. I originally had it using name for the lookup, but then realized that you can change the name of most objects when performing an update as long as the new one doesn't clash.

Reply:
Noooo, why should you ever rename something. It leads to problems. Yes, i know the api allows it, but we don't do it now.
Again: Don't focus on the api, but the admin usecase.
BTW: The change i am about to make will probably allow you to change the name. But still why...